components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -620,12 +640,42 @@ private Map<String, String> getUserClaimsInLocalDialect(String username,
                         null);
     }
 
+    private Map<String, String> getUserClaimsInLocalDialectWithUserID(String userId,
+                                                                      UserRealm realm,
+                                                                      List<String> claimURIList)
+            throws UserStoreException {
+
+        AbstractUserStoreManager userstore = (AbstractUserStoreManager) realm.getUserStoreManager();
+        if (userstore == null) {
+            throw new UserStoreException("Unable to retrieve UserStoreManager");
+        }
+        Map<String, String> userClaims =
+                userstore.getUserClaimValuesWithID(userId, claimURIList.toArray(new String[0]), null);
+        if (log.isDebugEnabled()) {
+            log.debug("User claims retrieved from user store: " + userClaims.size());
+        }
+        return userClaims;
+    }
+
+    private boolean isSubOrgLocalUserSwitchOrg(AuthenticatedUser authorizedUser) {
+
+        return "SSO".equals(authorizedUser.getFederatedIdPName()) && authorizedUser.isFederatedUser();
```

Comment:
We can't use `Stringutils.isNotEmpty(authorizedUser.getuserResidentOrganization())` since this would return true for both a local B2B user trying to switch organization and a federated B2B user trying to switch organizaion.

Reply:
Since we will be blocking any federated user without local association, we can use this check.

Reply:
Addressed with c202638bd7d45d70c3c4030d1f697325e8fdbc4f